### ISSUE TITLE
Bug, when nb_workers > nb_tasks with Memory File System

### ISSUE THREAD

**Issue by Xetd71** [nalepae/pandarallel#85]
**Problem description**
---------------------------------------------------------------------------
The process fails, when nb_workers > nb_tasks.

**Example:**
---------------------------------------------------------------------------
`pd.Series(np.arange(2)).parallel_apply(lambda x: x + 1)`

INFO: Pandarallel will run on 32 workers.
INFO: Pandarallel will use Memory file system to transfer data between the main process and `workers.`

**Error:**
---------------------------------------------------------------------------
```
IndexError                                Traceback (most recent call last)
<ipython-input-144-3c07ac8ca2ad> in <module>
----> 1 pd.Series(np.arange(2)).parallel_apply(lambda x: x + 1)

~/anaconda3/envs/py36/lib/python3.6/site-packages/pandarallel/pandarallel.py in closure(data, func, *args, **kwargs)
    449                 input_files,
    450                 output_files,
--> 451                 map_result,
    452             )
    453 

~/anaconda3/envs/py36/lib/python3.6/site-packages/pandarallel/pandarallel.py in get_workers_result(use_memory_fs, nb_workers, show_progress_bar, nb_columns, queue, chunk_lengths, input_files, output_files, map_result)
    375             if show_progress_bar:
    376                 progresses[worker_index] = chunk_lengths[worker_index]
--> 377                 progress_bars.update(progresses)
    378 
    379         elif message_type is ERROR:

~/anaconda3/envs/py36/lib/python3.6/site-packages/pandarallel/utils/progress_bars.py in update(self, values)
    116         """
    117         for index, value in enumerate(values):
--> 118             bar, label = self.__bars[index].children
    119 
    120             bar.value = value

IndexError: list index out of range
```

**Comment by nalepae** [comment]
Fixed in **Pandarallel v1.4.7**
Please reopen if you still encounter this issue.

**Comment by Xetd71** [comment]
Hello!
Now there is an error, if the parameter is set in `pandarallel.initialize` to: `use_memory_fs=True`:
```
---------------------------------------------------------------------------
EOFError                                  Traceback (most recent call last)
<ipython-input-8-3c07ac8ca2ad> in <module>
----> 1 pd.Series(np.arange(2)).parallel_apply(lambda x: x + 1)

~/anaconda3/envs/py36/lib/python3.6/site-packages/pandarallel/pandarallel.py in closure(data, func, *args, **kwargs)
    452                 input_files,
    453                 output_files,
--> 454                 map_result,
    455             )
    456 

~/anaconda3/envs/py36/lib/python3.6/site-packages/pandarallel/pandarallel.py in get_workers_result(use_memory_fs, nb_workers, show_progress_bar, nb_columns, queue, chunk_lengths, input_files, output_files, map_result)
    390     return (
    391         [pickle.load(output_files) for output_files in output_files]
--> 392         if use_memory_fs
    393         else results
    394     )

~/anaconda3/envs/py36/lib/python3.6/site-packages/pandarallel/pandarallel.py in <listcomp>(.0)
    389 
    390     return (
--> 391         [pickle.load(output_files) for output_files in output_files]
    392         if use_memory_fs
    393         else results

EOFError: Ran out of input
```

With parameter: `use_memory_fs=False` all works fine

**Comment by nalepae** [comment]
Thanks for you feedback,

Actually ran my tests on a macBook (without memory File System).

**Comment by nalepae** [comment]
Fixed in `Pandarallel v1.4.8` for Memory File System.
Please reopen if you still encounter this issue.
